src/docs/contribution_filter.md

```diff
@@ -0,0 +1,95 @@
+
+The Contribution Filter feature in Theia enables developers to remove specific contributions from a Theia-based application. This capability is typically used to eliminate existing features already present in Theia core, such as specific commands or menu entries.
+
+For a general understanding of contributions and contribution points, please refer to the [Services and Contributions](services_and_contributions/) documentation.
+
+## Overview
+
+The Contribution Filter feature provides a way to selectively disable certain contributions within a Theia application. This is handeled by the `ContributionFilterRegistry` and `FilterContribution`s  which allow you to register and apply filters to contributions.
+
+### FilterContribution API
+
+The `FilterContribution` interface allows you to register your contribution filters. These filters will be used by the `ContributionFilterRegistry` to remove or modify contributions.
+
+- **registerContributionFilters(registry: ContributionFilterRegistry): void**
+  - Use this method to register your contribution filters with the `ContributionFilterRegistry`.
+
+## Example Usage
+
+Here's an example of how to use the Contribution Filter to remove specific commands from a Theia application. In this example, we will first define some commands and the filter them. If you want to filter some existing contributions, you would of course skip the first step and just provide a `FilterContribution`.
+
+### Step 1: Define Example Commands and implement Command Contributions
+
+First, we define an example command that we want to filter out ([more information about commands](/commands_keybindings)).
+
+```typescript
+import { Command } from '@theia/core/lib/common';
+
+export namespace SampleFilteredCommand {
+
+    const EXAMPLE_CATEGORY = 'Examples';
+
+    export const TO_BE_FILTERED: Command = {
+        id: 'example_command.filtered',
+        category: EXAMPLE_CATEGORY,
+        label: 'This command should be filtered out'
+    };
+}
+```
+We register the example command with a command contribution.
+
+```typescript
+import { CommandContribution, CommandRegistry } from '@theia/core/lib/common';
+import { injectable } from 'inversify';
+
+@injectable()
+export class SampleFilteredCommandContribution implements CommandContribution {
+
+    registerCommands(commands: CommandRegistry): void {
+        commands.registerCommand(SampleFilteredCommand.TO_BE_FILTERED, { execute: () => { } });
+    }
+}
+```
+
+### Step 2: Register Filters
+
+Now, we register the filters to remove specific contributions, in our case the example command we just registered.
+
+```typescript
+import { FilterContribution, ContributionFilterRegistry, CommandContribution } from '@theia/core/lib/common';
+import { injectable } from 'inversify';
+
+@injectable()
+export class SampleFilterContribution implements FilterContribution {
+
+    registerContributionFilters(registry: ContributionFilterRegistry): void {
+        registry.addFilters('*', [
```

Comment:
I would opt for scoping the filter in the example to be only applied on command contributions.
Using scoped filters is preferable as it ensures that the filter only is applied when necessary and in addition
this avoids filtering out of "false positives" in other contributions. (Not likely to happen with the instance of based filter in the example, but with arbitrary filters this could occur)

Maybe then also a short setence to mention that it's possible to register global filters using the `*` operator.
```suggestion
        registry.addFilters([CommandContribution], [
```